### PR TITLE
Enable travis to check the build, tests and formatting

### PR DESCRIPTION
**Pull request by dtebbs**
This PR sets up travis to check code formatting, check the build (including compile warnings) and run some fast-running tests on mac and linux.

Addresses issue #60.

All builds execute with "-Werror" to detect compiler warnings.
The linux build pulls the docker container and runs build and tests inside that.  It runs the code checks natively (relying on git to notice any changes).  The mac build uses the default aple compilers.  In the future it would be nice to use the llvm compilers on mac.

### CODE REVIEW

**Comment by dtebbs** [comment]
If there are other configs we should be testing then let me know.

File: scripts/ci [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    num_lines=`cat format_errors | wc -l`
+    if [ "0" != "${num_lines}" ] ; then
+        echo CODE FORMATTING ERRORS:
+        cat format_errors
+        exit 1
+    fi
+}
+
+function build() {
+
```
Comment (AntoineRondelet) [inline scripts/ci]: Extra line

File: scripts/ci [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        export PATH="/usr/local/opt/llvm/bin:/usr/local/bin:${PATH}"
+        export PKG_CONFIG_PATH="/usr/local/opt/openssl/lib/pkgconfig"
+        export LIBRARY_PATH="/usr/local/opt/openssl/lib"
+        export LDFLAGS="-L/usr/local/opt/llvm/lib -L-L/usr/local/lib"
+        export CPPFLAGS="-I/usr/local/opt/llvm/include -I/usr/local/include"
+
+        cxx_flags="${cxx_flags} -Wno-deprecated-declarations"
+    fi
+
+    cmake_flags="-DFAST_TESTS_ONLY=ON -DCMAKE_BUILD_TYPE=${build_type}"
```
Comment (AntoineRondelet) [inline scripts/ci]: Why skip the prover tests? It's a slow one but this is the one that tests the entire circuit.

File: src/test/simple_test.tcc [hunk 3]
```diff
@@ -32,19 +32,19 @@ template<typename FieldT> void simple_circuit(libsnark::protoboard<FieldT> &pb)
     pb.set_input_sizes(1);
 
     //   g1
-    //  /  \
+    //  /  \ .
```
Comment (AntoineRondelet) [inline src/test/simple_test.tcc]: Why did you add `.`?

File: scripts/ci (hunk 2 above)
Comment (AntoineRondelet) [reply scripts/ci]: We want to make sure all tests pass before merging

File: scripts/ci (hunk 1 above)
Comment (dtebbs) [reply scripts/ci]: Removed extra line

File: scripts/ci (hunk 2 above)
Comment (dtebbs) [reply scripts/ci]: I've enabled the prover_test as well.
I originally disabled it for these per-commit builds to get faster turnaround time on results, and wasn't sure if travis had some timeout.  But let's see how it goes.  Definitely better if we can run it on every check-in.

File: src/test/simple_test.tcc (hunk 3 above)
Comment (dtebbs) [reply src/test/simple_test.tcc]: Yeah, this is quite annoying.  One of the compilers was giving warnings about multi-line comments if there was a `\` at the end of a comment.  I could either try to disable that warning, or adjust the form of the comments if you prefer.

Comment (dtebbs) [reply src/test/simple_test.tcc]: Afterall, I changed the comments to not use `\` at the end.

File: scripts/ci (hunk 2 above)
Comment (dtebbs) [reply scripts/ci]: prover_test is enabled and running now (builds take approx 22 mins).

Comment (AntoineRondelet) [reply scripts/ci]: Yes, let's see how it goes. Thanks!
